File: php/features.php
<?php
declare(strict_types=1);

// NewsPublic SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class NewsPublicFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new NewsPublicBaseFeature();
            case "test":
                return new NewsPublicTestFeature();
            default:
                return new NewsPublicBaseFeature();
        }
    }
}

File: py/newspublic_sdk/features.py
# NewsPublic SDK feature factory

from newspublic_sdk.feature.base_feature import NewsPublicBaseFeature
from newspublic_sdk.feature.test_feature import NewsPublicTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NewsPublicBaseFeature(),
        "test": lambda: NewsPublicTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
